controllers/imagepolicy_controller.go

```diff
@@ -164,3 +183,28 @@ func (r *ImagePolicyReconciler) imagePoliciesForRepository(obj handler.MapObject
 	}
 	return reqs
 }
+
+// event emits a Kubernetes event and forwards the event to notification controller if configured
+func (r *ImagePolicyReconciler) event(policy imagev1alpha1.ImagePolicy, severity, msg string) {
+	if r.EventRecorder != nil {
+		r.EventRecorder.Eventf(&policy, "Normal", severity, msg)
+	}
+	if r.ExternalEventRecorder != nil {
+		objRef, err := reference.GetReference(r.Scheme, &policy)
+		if err != nil {
```

Comment:
Oh for an error monad. Can you collapse these two identical error-handling clauses into one? I use

```
err := operation1()
if err == nil {
  err = operation2()
}
if err != nil {
  // handle error
}
```

but some people find that distasteful (and being honest, I have tripped myself up with it in the past :-/)